semantic-conventions/src/opentelemetry/semconv/templating/markdown/__init__.py

```diff
@@ -416,31 +443,36 @@ def _render_group(self, semconv, parameters, output):
         self.render_ctx.is_remove_constraint = "remove_constraints" in parameters
         self.render_ctx.group_key = parameters.get("tag")
         self.render_ctx.is_full = "full" in parameters
+        self.render_ctx.is_metric_table = "metric_table" in parameters
 
         if isinstance(semconv, EventSemanticConvention):
             output.write("The event name MUST be `{}`.\n\n".format(semconv.name))
 
         attr_to_print = []
-        attr: SemanticAttribute
-        for attr in sorted(
-            semconv.attributes, key=lambda a: "" if a.ref is None else a.ref
-        ):
-            if self.render_ctx.group_key is not None:
-                if attr.tag == self.render_ctx.group_key:
+        if self.render_ctx.is_metric_table:
+            self.to_markdown_metric_table(semconv, output)
+        else:
+            attr: SemanticAttribute
```

Comment:
Extract the whole `else` branch to a function named `to_markdown_attribute_table`? Would probably also make the diff look nicer because the indentation would not change then.

Reply:
I thought about this initially but there was a soup of variables used inside and out of the branch. Anyway, I had a closer look and went ahead and did it. Things just needed a little re-arranging.